conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/FrameworkException.java

```diff
@@ -25,23 +27,31 @@
 /** Internal type to signal a conjure protocol-level failure with a specific response code. */
 final class FrameworkException extends RuntimeException implements SafeLoggable {
 
+    private static final ErrorType UNPROCESSABLE_ENTITY = ErrorType.create(
+            ErrorType.Code.INVALID_ARGUMENT, "Conjure:UnprocessableEntity");
+    private static final ErrorType UNSUPPORTED_MEDIA_TYPE = ErrorType.create(
+            ErrorType.Code.INVALID_ARGUMENT, "Conjure:UnsupportedMediaType");
+
     private final String logMessage;
     private final List<Arg<?>> arguments;
     private final int statusCode;
+    private final ErrorType errorType;
 
-    private FrameworkException(String message, int statusCode, Throwable cause, Arg<?>... args) {
+    private FrameworkException(String message, ErrorType errorType, int statusCode, Throwable cause, Arg<?>... args) {
         super(renderMessage(message, args), cause);
         this.logMessage = message;
         this.arguments = ImmutableList.copyOf(args);
         this.statusCode = statusCode;
+        this.errorType = errorType;
     }
 
-    static FrameworkException unprocessableEntity(String message, Throwable cause, Arg<?>... args) {
-        return new FrameworkException(message, StatusCodes.UNPROCESSABLE_ENTITY, cause, args);
+    static FrameworkException unprocessableEntity(
+            @CompileTimeConstant String message, Throwable cause, Arg<?>... args) {
+        return new FrameworkException(message, UNPROCESSABLE_ENTITY, StatusCodes.UNPROCESSABLE_ENTITY, cause, args);
```

Comment:
we should probably get the error code out of the `ErrorType.Code` directly rather than potentially creating a mismatch

Reply:
FrameworkException is an internal detail of conjure-undertow to support failures which do not have a status code in ErrorType.Code.

Reply:
Never mind, I guess you can't really do that since that enum doesn't define a code for 412 nor 415 🙄